File: LeetCode/121-Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = prices[0], max = prices[0];
        int ans = 0;
        for (int i = 0; i < prices.length; i++) {
            if (prices[i] < min)
                min = prices[i];
            else {
                max = prices[i];
                ans = Math.max(ans, max - min);
            }
        }
        return ans;
    }
}

File: LeetCode/121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& arr) 
    {
        int profit=0,maxprofit=0,n=arr.size(),minsofar=arr[0];
        for (int i = 0; i <n; i++) 
        {
            minsofar = min(minsofar,arr[i]);
            maxprofit = arr[i] - minsofar;
            profit = max(profit,maxprofit);  
        }
    return profit;
    }
};
